exercises/src/main/scala/fpinscala/datastructures/List.scala

```diff
@@ -21,7 +21,7 @@ object List { // `List` companion object. Contains functions for creating and wo
 
   def apply[A](as: A*): List[A] = // Variadic function syntax
     if (as.isEmpty) Nil
-    else Cons(as.head, apply(as.tail: _*))
+    else Cons(as.head, apply(as.tail: _*)) // how come this one works? .tail?
```

Comment:
I had to replace all the x.tail method calls with function calls tail(x) - not sure why for their code, this .tail doesn't cause an error?